src/main/java/org/advent/dayeight/DayEight.java

```diff
@@ -0,0 +1,119 @@
+package org.advent.dayeight;
+
+import org.advent.PrintSolution;
+import org.advent.Util;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DayEight implements PrintSolution {
+
+    public Map<String, Node> nodeMap(List<String> listOfStrings) {
+        return listOfStrings.stream()
+                .collect(Collectors.toMap(
+                        s -> s.substring(0,3),
+                        s -> new Node(s.substring(0, 3), s.substring(7, 10), s.substring(12,15))
+                        ));
+    }
+
+    public Node findNodeMapFromPointer(Map<String, Node> nodeMap, Node node, char direction) {
+        if (direction == 'L') {
+            return nodeMap.get(node.leftPointer());
+        } else {
+            return nodeMap.get(node.rightPointer());
+        }
+    }
+
+    public long numberOfStepsToReachZZZ(List<String> listOfStrings, String directions) {
+        Map<String, Node> nodeMap = nodeMap(listOfStrings);
+        String startingAddress = "AAA";
+        Node currentNode = nodeMap.get(startingAddress);
+
+        return numberOfStepsToFindString(listOfStrings, currentNode, directions, "ZZZ");
+    }
+
+    public Long LCMOfAllPaths(List<String> listOfStrings, String directions) {
+        Map<String, Node> filteredMapStarting = getFilteredMap(nodeMap(listOfStrings));
+        List<Node> listOfNodes = new ArrayList<>(filteredMapStarting.values());
+        List<Long> numberOfStepsList = new ArrayList<>();
+
+        for (Node currentNode : listOfNodes) {
+            numberOfStepsList.add(numberOfStepsToFindString(listOfStrings, currentNode, directions, "Z"));
+        }
+        long lcmOfAllValues = lcm(numberOfStepsList.get(0), numberOfStepsList.get(1));
+        if (numberOfStepsList.size() > 2) {
+            for (int i = 2; i < numberOfStepsList.size(); i++) {
+                lcmOfAllValues = lcm(lcmOfAllValues, numberOfStepsList.get(i));
+            }
+        }
+        return lcmOfAllValues;
+    }
+
+    @Override
+    public void printPartOne() {
+        System.out.println(" ");
+        System.out.println(this.getClass().getSimpleName() + " ---------------------------");
+        System.out.println("Total number of steps for part one = " + results(true));
+    }
+
+    @Override
+    public void printPartTwo() {
+        System.out.println(" ");
+        System.out.println(this.getClass().getSimpleName() + " ---------------------------");
+        System.out.println("Total number of steps for part two = " + results(false));
+    }
+
+    private long numberOfStepsToFindString(List<String> listOfStrings, Node currentNode, String directions, String stringToFind) {
+        int numberOfSteps = 0;
```

Comment:
why is this not a long? 
this is a serious fault in your code. You say in the method signature that you expect a long to be the answer of this, but number of steps is just an int, and then you autocast it to long during the return. if the number of steps is > Integer.MAXVALUE then you will loop around and start at min value again but never return the correct result